internal/kafka/internal/workers/kafka_mgrs/promotion/chain/reconciler_chain.go

```diff
@@ -0,0 +1,55 @@
+package chain
```

Comment:
Is there some reason this has been implemented with generics over just interfaces?

I suspect it is due to having potentially different return types in the results between reconcile actions.

What would be the alternative without using generics? I think it is good we evaluate the pros/cons of using each approach

Reply:
> Yes, that is the reason. Different chains could produce different results. Using an interface, we would end up using the any value.

If `any` is used with interface that means that the receiver of the result should perform downcasting to get the real type right?

Reply:
yes